crates/ruff_linter/src/rules/refurb/rules/single_item_membership_test.rs

```diff
@@ -100,12 +98,8 @@ pub(crate) fn single_item_membership_test(
         expr.range(),
     );
 
-    let applicability =
-        if right.is_string_literal_expr() || checker.comment_ranges().intersects(expr.range()) {
-            Applicability::Unsafe
-        } else {
-            Applicability::Safe
-        };
+    // All supported cases can change runtime behavior; mark as unsafe.
+    let applicability = Applicability::Unsafe;
```

Comment:
Previously, fixes were marked unsafe only for string literals and when comments intersected the edit range, but single‑item list/tuple/set/frozenset cases can also
  change behavior (e.g., NaN identity differences and non‑boolean eq), so we now mark all FURB171 fixes as unsafe.